engine-tests/src/test/java/org/terasology/rendering/particles/internal/ParticlePoolTest.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright 2015 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.particles.internal;
+
+import org.junit.Test;
+import org.terasology.math.TeraMath;
+import org.terasology.rendering.particles.internal.data.ParticlePool;
+import org.terasology.utilities.random.FastRandom;
+import org.terasology.utilities.random.Random;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+/**
+ * Created by Linus on 28-2-2015.
```

Comment:
Terasology doesn't use author headers - please remove these (or replace with detailed and useful Javadoc, wink wink nudge nudge 😉)